lib/backend/etcdv3/watcher.go

```diff
@@ -109,36 +73,18 @@ func (wc *watcher) ResultChan() <-chan api.WatchEvent {
 	return wc.resultChan
 }
 
-// listCurrent retrieves the existing entries and sends an event for each listed
-func (wc *watcher) listCurrent() error {
-	log.Info("Performing initial list with no revision")
-	list, err := wc.client.List(wc.ctx, wc.list, "")
-	if err != nil {
-		return err
-	}
-
-	wc.initialRev, err = strconv.ParseInt(list.Revision, 10, 64)
-	if err != nil {
-		log.WithError(err).Error("List returned revision that could not be parsed")
-		return err
-	}
-
-	// We are sending an initial sync of entries to the watcher to provide current
-	// state.  To the perspective of the watcher, these are added entries, so set the
-	// event type to WatchAdded.
-	for _, kv := range list.KVPairs {
-		log.Info("Sending create events for each existing entry")
-		wc.sendEvent(&api.WatchEvent{
-			Type: api.WatchAdded,
-			New:  kv,
-		})
-	}
-	return nil
+// HasTerminated implements the api.WatchInterface.
+func (wc *watcher) HasTerminated() bool {
+	terminated := wc.terminated
```

Comment:
cant you just do `return wc.terminated`?

Reply:
ha!  yes.  I think I previously had a log in there too, and I want to make sure that the log and the return value were in-sync.

Reply:
I'm gonna change this to use an atomic operator.